pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -454,15 +454,16 @@ func (ds *DiscoveryService) doPush() {
 	// more config update events may happen while doPush is processing.
 	// we don't want to lose updates.
 	clearCacheMutex.Lock()
-
 	clearCacheTimerSet = false
 	lastClearCache = time.Now()
 	full := ds.fullPush
+	clearCacheMutex.Unlock()
+
 	edsUpdates := BeforePush()
```

Comment:
Can we put it before lock? So we no need lock twice

Reply:
Not sure what you mean, more details please.